qurro/qarcoal.py

```diff
@@ -80,8 +84,11 @@ def filter_and_join_taxonomy(feat_table, taxonomy, num_string, denom_string):
         raise ValueError(
             "No samples contain both numerator and denominator features!"
         )
-    tax_num_df = tax_num_df[samp_to_keep]
-    tax_denom_df = tax_denom_df[samp_to_keep]
+
+    # convert to dense to avoid issues with sparse data structures
+    # because these are filtered, sparsity shouldn't be a concern anyway
+    tax_num_df = tax_num_df[samp_to_keep].sparse.to_dense()
+    tax_denom_df = tax_denom_df[samp_to_keep].sparse.to_dense()
```

Comment:
I don't *think* we can rule out sparsity not being a problem here -- there's always going to be the corner case bogeyman of "samples that contain counts of only the numerator and denominator features and have 0s everywhere else" to worry about.

...I doubt that'll be a problem for like 99% of users, but it is a factor in favor of not using dense structures for this sort of data wherever possible.